components/speedreader/resources/panel/main-container.tsx

```diff
@@ -0,0 +1,65 @@
+import * as React from 'react'
+
+import { panelDataHandler, SiteSettings, Theme, FontSize, ContentStyle, FontFamily } from './api/browser'
+import MainPanel from './components/main-panel'
+
+function MainContainer () {
+  const [siteSettings, setSiteSettings] = React.useState<SiteSettings | undefined>()
+
+  React.useEffect(() => {
+    const onVisibilityChange = () => {
+      if (document.visibilityState === 'visible') {
+        panelDataHandler.getSiteSettings().then(res => setSiteSettings(res.siteSettings))
+      }
+    }
+
+    document.addEventListener('visibilitychange', onVisibilityChange)
+    panelDataHandler.getSiteSettings().then(res => setSiteSettings(res.siteSettings))
+
+    return () => {
+      document.removeEventListener('visibilitychange', onVisibilityChange)
+    }
+  }, [])
+
+  if (!siteSettings) {
+    return null
+  }
+
+  const handleThemeChange = (theme: Theme) => {
+    setSiteSettings({ ...siteSettings, theme })
+    panelDataHandler.setTheme(theme)
+  }
+
+  const handleFontSizeChange = (fontSize: FontSize) => {
+    setSiteSettings({ ...siteSettings, fontSize })
+    panelDataHandler.setFontSize(fontSize)
+  }
+
+  const handleContentStyleChange = (contentStyle: ContentStyle) => {
+    setSiteSettings({ ...siteSettings, contentStyle })
+    panelDataHandler.setContentStyle(contentStyle)
+  }
+
+  const handleFontFamilyChange = (fontFamily: FontFamily) => {
+    setSiteSettings({ ...siteSettings, fontFamily })
+    panelDataHandler.setFontFamily(fontFamily)
+  }
+
+  const handleToggleChange = (isOn: boolean) => {
+    panelDataHandler.setEnabled(isOn)
+    panelDataHandler.getSiteSettings().then(res => setSiteSettings(res.siteSettings))
+  }
+
+  return (
+   <MainPanel
```

Comment:
nit: `MainPanel` and `MainContainer` are pretty similar. How about `SpeedreaderPanel` and `Container`. Not sure if that's better. Alternatively use a hook for this stuff, or context.

Reply:
renaming to Container works better for now.